torch/distributed/fsdp/_fully_shard/_fully_shard.py

```diff
@@ -212,7 +212,7 @@ def fully_shard(
     managed_modules = _get_managed_modules(modules, ignored_params)
     params, buffers = _get_managed_states(managed_modules, ignored_params)
 
-    _move_states_to_device(params, buffers, device)
+    _move_states_to_device(params, buffers, ignored_params, device)
```

Comment:
I would recommend including in the docstring part for `ignored_params` arg to mention that they will still be moved to device if needed.